tests/e2e/upgradecluster/upgradecluster_test.go

```diff
@@ -0,0 +1,412 @@
+					g.Expect(res).Should((ContainSubstring("test-nodeport")), func() string { return res })
+				}, "420s", "5s").Should(Succeed())
+
+			}
+		})
+
+		It("\nAfter upgrade validating Service LoadBalancer", func() {
+			for _, element := range serverNodeNames {
+				nodeName := strings.TrimSpace(element)
+				if nodeName == "" {
+					continue
+				}
+				ip, _ := e2e.FetchNodeExternalIP(nodeName)
+				cmd := "kubectl get service nginx-loadbalancer-svc --kubeconfig=" + kubeConfigFile + " --output jsonpath=\"{.spec.ports[0].port}\""
+				port, _ := e2e.RunCommand(cmd)
+				cmd = "curl -L --insecure http://" + ip + ":" + port + "/name.html"
+				fmt.Println(cmd)
+				res, _ := e2e.RunCommand(cmd)
+				fmt.Println(res)
+				Eventually(func(g Gomega) {
+					cmd := "kubectl get pods -o=name -l k8s-app=nginx-app-loadbalancer --field-selector=status.phase=Running --kubeconfig=" + kubeConfigFile
+					res, _ := e2e.RunCommand(cmd)
+					g.Expect(res).Should((ContainSubstring("test-loadbalancer")), func() string { return res })
+				}, "420s", "5s").Should(Succeed())
+			}
+		})
+		It("After upgrade validating Ingress", func() {
+			for _, element := range serverNodeNames {
+				nodeName := strings.TrimSpace(element)
+				if nodeName == "" {
+					continue
+				}
+				ip, _ := e2e.FetchNodeExternalIP(nodeName)
+				cmd := "curl  --header host:foo1.bar.com" + " http://" + ip + "/name.html"
+				fmt.Println(cmd)
+
+				Eventually(func(g Gomega) {
+					res, _ := e2e.RunCommand(cmd)
+					fmt.Println(res)
+					g.Expect(res).Should((ContainSubstring("test-ingress")), func() string { return res })
+				}, "420s", "5s").Should(Succeed())
+			}
+		})
+
+		It("\nAfter upgrade validating Daemonset", func() {
+			nodes, _ := e2e.ParseNodes(kubeConfigFile, false) //nodes :=
+			pods, _ := e2e.ParsePods(kubeConfigFile, false)
+
+			Eventually(func(g Gomega) {
+				count := e2e.CountOfStringInSlice("test-daemonset", pods)
+				fmt.Println("POD COUNT")
+				fmt.Println(count)
+				fmt.Println("NODE COUNT")
+				fmt.Println(len(nodes))
+				g.Expect(len(nodes)).Should((Equal(count)), func() string { return "count" })
+			}, "420s", "1s").Should(Succeed())
+		})
+		It("After upgrade validating dns access", func() {
+			Eventually(func(g Gomega) {
+				cmd := "kubectl --kubeconfig=" + kubeConfigFile + " exec -i -t dnsutils -- nslookup kubernetes.default"
+				fmt.Println(cmd)
+				res, _ := e2e.RunCommand(cmd)
+				fmt.Println(res)
+				g.Expect(res).Should((ContainSubstring("kubernetes.default.svc.cluster.local")), func() string { return res })
+			}, "180s", "2s").Should(Succeed())
+		})
+		It("After upgrade validating Local Path Provisioner storage ", func() {
+			Eventually(func(g Gomega) {
+				cmd := "kubectl exec volume-test cat /data/test --kubeconfig=" + kubeConfigFile
+				res, _ := e2e.RunCommand(cmd)
+				fmt.Println("Data after re-creation", res)
+				g.Expect(res).Should((ContainSubstring("local-path-test")), "420s", "5s", func() string { return res })
```

Comment:
Why do we have two sets of waiting loops here? 
What is the `func() string {return res})` doing